python/cuml/linear_model/ridge_mg.pyx

```diff
@@ -0,0 +1,406 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+# cython: language_level = 3
+
+import ctypes
+import cudf
+import numpy as np
+
+import rmm
+
+from libc.stdlib cimport malloc, free
+
+from libcpp cimport bool
+from libc.stdint cimport uintptr_t, uint32_t, uint64_t
+from cython.operator cimport dereference as deref
+
+from cuml.common.base import Base
+from cuml.common.handle cimport cumlHandle
+from cuml.decomposition.utils cimport *
+from cuml.utils import get_cudf_column_ptr, get_dev_array_ptr, \
+    input_to_dev_array, zeros
+
+from cuml.linear_model import Ridge
+
+cdef extern from "cumlprims/opg/matrix/data.hpp" \
+                 namespace "MLCommon::Matrix":
+
+    cdef cppclass floatData_t:
+        floatData_t(float *ptr, size_t totalSize)
+        float *ptr
+        size_t totalSize
+
+    cdef cppclass doubleData_t:
+        doubleData_t(double *ptr, size_t totalSize)
+        double *ptr
+        size_t totalSize
+
+cdef extern from "cumlprims/opg/matrix/part_descriptor.hpp" \
+                 namespace "MLCommon::Matrix":
+
+    cdef cppclass RankSizePair:
+        int rank
+        size_t size
+
+cdef extern from "cumlprims/opg/ridge.hpp" namespace "ML::Ridge::opg":
+
+    cdef void fit(cumlHandle& handle,
+		  RankSizePair **rank_sizes,
+		  size_t n_parts,
+		  floatData_t **input,
+		  size_t n_rows,
+		  size_t n_cols,
+		  floatData_t **labels,
+                  float *alpha,
```

Comment:
indent issues with these two lines and algo below.

Reply:
Should be fixed already because flake8 is not complaining.